doc/contributing.md

```diff
@@ -1,16 +1,171 @@
+* refactoring: refactoring can be a good way to improve a code base.
+
+### Developer Instructions
+
+
+```{note}
+The HoloViz projects usually provide specific instructions to set up your environment to be ready to contribute documentation or code to the project. The instructions provided hereafter are meant to be general enough to apply to each project. Refer to the documentation of the project you are working on for more details.
+```
+
+Before making any contribution, the first step is usually to open a [GitHub Issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) that documents what you are trying to fix/improve. Once an issue is opened, a discussion can take place with the maintainers to see if what is suggested is relevant, which can orientate the action to take.
+
+The first step to working on any source code is to install Git as the source codes are stored in [Git](https://git-scm.com) source control repositories. To install Git on any platform, refer to the [Installing Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) section of the [Pro Git Book](https://git-scm.com/book/en/v2).
+
+As an external contributor, you do not have permission to submit code directly to any of the repositories. Github offers you to fork the repository, which will create a copy of the repository, on which you will have the right to work freely. Follow [these instructions](https://docs.github.com/en/get-started/quickstart/fork-a-repo) to find out how to fork and clone a repository.
+
+The HoloViz developers rely on [Conda](https://conda.io/docs/intro.html) to manage their virtual environments. We **recommend** you install Conda too to have an experience as close as possible to those of the core developers. To install Conda on any platform, you can [install Miniconda](https://docs.conda.io/en/latest/miniconda.html). 
+
+The HoloViz developers rely on [pyctdev](https://github.com/pyviz-dev/pyctdev), a custom developer tool that facilitates maintaining all of the holoviz projects. pyctdev is an extension of the task running tool [doit](https://pydoit.org/). We **recommend** that you install `pyctdev`:
+
+```bash
+conda install -c pyviz/label/dev "pyctdev>0.5.0"
+```
+
+Once `pyctdev` is available and you are in the cloned panel repository, you can create a new virtual environment with the `env_create` *doit* command with the appropriate channels:
+
+```bash
+doit env_create -c channel1 -c channel2 --name=dev-env --python=3.x
+```
+
+Don't forget to activate this environment:
+
+```bash
+conda activate dev-env
+```
+
+To perform an editable install of the project you are working on, including all the dependencies required to run the full unit test suite, run the `develop_install` with the appropriate channels and options:
+
+```bash
+doit develop_install -c channel1 -c channel2 -o tests -o ... -o ...
```

Comment:
maybe add an example ~ ('For example: `doit -c pyviz -c conda-forge develop_install -o tests -o examples`')